### PR TITLE
Documentation and Config updates

### PR DESCRIPTION
**Pull request by jaley**
This PR finished up the V3 release by adding docs for new functionality, removing irrelevant and superseded docs and config and fixing the local docker build.

@paddybyers and @tomczoink -- I added you because the README changes give you a decent summary of what's new and thought it might be good for you to review that.

### CODE REVIEW

**Review by AndyTWF — APPROVED** [review]
From a high level this looks good - will wait on a review from someone more familiar with the changes before I merge it :)

**Review by lmars — APPROVED** [review]

File: README.md
Comment (tomczoink) [inline README.md:221]: I believe the context for patterns is in `Dynamic Channel Configuration`, with channel just directing to that section, so we could just direct to that section instead.

File: README.md
Comment (tomczoink) [inline README.md:222]: For all of these limits are there any additional maximums and minimums to what these values can be?

File: README.md
Comment (tomczoink) [inline README.md:304]:
```suggestion
has `STRUCT` type and `userId` can be converted to a string. Given that both message and channel names are ultimately
```

File: README.md
Comment (tomczoink) [inline README.md:359]:
```suggestion
* Failed dynamic channel mappings, if `onFailedRecordMapping = dlq`
```

File: src/main/java/com/ably/kafka/connect/config/ChannelSinkConnectorConfig.java
Comment (tomczoink) [inline src/main/java/com/ably/kafka/connect/config/ChannelSinkConnectorConfig.java:139]: This is 100, but in https://github.com/ably/kafka-connect-ably/pull/142/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R223 (which I believe is what this is the default value for) it's still 1000.

File: src/main/java/com/ably/kafka/connect/config/ChannelSinkConnectorConfig.java
Comment (tomczoink) [inline src/main/java/com/ably/kafka/connect/config/ChannelSinkConnectorConfig.java:144]: This is 100, but in https://github.com/ably/kafka-connect-ably/pull/142/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R224 (which I believe this is the default value for) it's still 5000.

**Review by tomczoink — CHANGES_REQUESTED** [review]

File: README.md
Comment (AndyTWF) [reply README.md:222]: Done in https://github.com/ably/kafka-connect-ably/pull/142/commits/aba1da6aab4ddd28ec80d4ef51080c3195a73a26

File: src/main/java/com/ably/kafka/connect/config/ChannelSinkConnectorConfig.java
Comment (AndyTWF) [reply src/main/java/com/ably/kafka/connect/config/ChannelSinkConnectorConfig.java:139]: Done in https://github.com/ably/kafka-connect-ably/pull/142/commits/83672a0b3ba3ea277f88d938d50ac9743e2021e7

File: src/main/java/com/ably/kafka/connect/config/ChannelSinkConnectorConfig.java
Comment (AndyTWF) [reply src/main/java/com/ably/kafka/connect/config/ChannelSinkConnectorConfig.java:144]: Done in https://github.com/ably/kafka-connect-ably/pull/142/commits/83672a0b3ba3ea277f88d938d50ac9743e2021e7

File: README.md
Comment (AndyTWF) [reply README.md]: Done in https://github.com/ably/kafka-connect-ably/pull/142/commits/aba1da6aab4ddd28ec80d4ef51080c3195a73a26

File: README.md
Comment (AndyTWF) [reply README.md:222]: For batch publishing, you'd be limited by the maximum payload size limits here:

https://ably.com/docs/api/rest-api#batch-publish

So it would be a case-by-case basis

**Review by tomczoink — APPROVED** [review]
Looks good, thanks!

File: README.md
Comment (AndyTWF) [reply README.md:222]: Added a line about the limits: https://github.com/ably/kafka-connect-ably/pull/142/commits/162248ad6fedc2647f445a6e19b175b6740c69a9

**Pull request merged**
